README.md

```diff
@@ -0,0 +1,124 @@
+
+> [!NOTE]
+> Ensure you have Java 21 or later installed on your machine.
+
+### `hide`
+
+The `hide` command allows you to hide any file inside a bitmap image. The syntax is as follows:
+
+```shell
+java -jar shadow-1.0.0.jar <path-to-bitmap-image> <path-to-file-to-hide> hide <path-to-output-bmp-image>
+```
+
+Where `<path-to-bitmap-image>` is the path to the bitmap image,`<path-to-file-to-hide>` is the path to the file you want to hide and `<path-to-output-bmp-image>` is the path where the bitmap image with the hidden content will be available.
+
+You can also specify how many bits per byte of the image you want to use to hide the file. The default value is 1. To specify the number of bits per byte, you can use the `-b`/`--bits-per-byte` flag:
+
+```shell
+java -jar shadow-1.0.0.jar <path-to-bitmap-image> <path-to-file-to-hide> hide <path-to-output-bmp-image> -b <number-of-bits-per-byte>
+```
+
+> [!NOTE]
+> The number of bits per byte must be between 1 and 8 and a power of 2.
+
+> [!WARNING]
+> The more bits per byte you use, the more noticeable the changes in the image will be, since the changes in the colors of the pixels will be more significant.
+
+If a file with the same name as the output file already exists, the program will ask you if you want to overwrite it. You can do that by re-executing the command with the `-f`/`--force` flag before the subcommand `hide`:
+
+```shell
+java -jar shadow-1.0.0.jar <path-to-bitmap-image> <path-to-file-to-hide> --force hide <path-to-output-bmp-image>
+```
+
+### `expose`
+
+The `expose` command allows you to retrieve the file hidden in a bitmap image. The syntax is as follows:
+
+```shell
+java -jar shadow-1.0.0.jar <path-to-bitmap-image-with-hidden-file> <path-to-output-file> expose
+```
+
+Where `<path-to-bitmap-image-with-hidden-file>` is the path to the bitmap image with the hidden file and `<path-to-output-file>` is the path where the hidden file will be saved.
+
+If the bitmap image provided does not contain a hidden file, the program will return an error message. Also, if the output file already exists, the program will ask you if you want to overwrite it. You can do that by re-executing the command with the `-f`/`--force` flag before the subcommand `expose`:
+
+```shell
+java -jar shadow-1.0.0.jar <path-to-bitmap-image-with-hidden-file> <path-to-output-file> --force expose
+```
+
+> [!IMPORTANT]
+> Since the storage of the bits of an hidden file overwrites the original bits of the image, the retrieval of the hidden file will set the bits of the image to 0 and not their original value.
+
+<!-- TODO Add Docker examples after first release -->
+
+## Demonstration
+
+Here we will show a demonstration of the `hide` and `expose` commands using the files available in the [`examples`](./examples/) directory.
+
+We want to hide this small video:
+
+![Video to hide](https://github.com/user-attachments/assets/5e3d14c7-d8aa-4b4c-8539-167b9e1eb6a3)
```

Comment:
We should put the source somewhere. Source: https://sample-videos.com/video321/mp4/720/big_buck_bunny_720p_5mb.mp4
Note: website certificate expired on October 2nd 2024. There's a risk it might become unavailable soon

Reply:
Done!